### PR TITLE
Schedule name shouldn't contains special character 

### PR DESCRIPTION
**Pull request by ravydv**
## Checklist

- [x] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [ ] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [ ] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [ ] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
        - {Please put the link here}
    - [ ] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

**Comment by jagpreetstamber** [comment]
Approved

**Comment by jagpreetstamber** [comment]
#10063

File: src/Automation/Automation/Cmdlet/NewAzureAutomationSchedule.cs [hunk 1]
```diff
@@ -149,6 +149,11 @@ public NewAzureAutomationSchedule()
         [PermissionSet(SecurityAction.Demand, Name = "FullTrust")]
         protected override void AutomationProcessRecord()
         {
+            if (!this.IsScheduleNameValid())
+            {
+                throw new ArgumentException(Resources.ScheduleNameInvalid);
```
Comment (isra-fel) [inline src/Automation/Automation/Cmdlet/NewAzureAutomationSchedule.cs:154]: Suggest using common exception type [AzPSArgumentException](https://github.com/Azure/azure-powershell-common/blob/main/src/Common/Exceptions/AzPSArgumentException.cs) for more precise telemetry data
```suggestion
                throw new AzPSArgumentException(Resources.ScheduleNameInvalid, nameof(Name));
```

**Review by isra-fel — CHANGES_REQUESTED** [review]
Please also update ChangeLog.md file - see the checklist in PR template for more info

**Review by isra-fel — APPROVED** [review]
LGTM

**Pull request merged**